overlord/snapstate/handlers.go

```diff
@@ -3988,9 +3988,9 @@ func changeReadyUpToTask(task *state.Task) bool {
 // in the last batch of refreshes before the given (re-refresh) task; failed is
 // true if any of the snaps failed to refresh.
 //
-// It does this by advancing through the given task's change's tasks, keeping
-// track of the instance names from the first SnapSetup in every lane, stopping
-// when finding the given task, and resetting things when finding a different
+// It does this by advancing through the given task's change's tasks, and keeping
+// track of the instance names from every SnapSetup it finds.
```

Comment:
SnapSetup for a download-snap task it finds?

Reply:
This is a good improvement, added this